src/Compilers/CSharp/Portable/BoundTree/BoundExpression.cs

```diff
@@ -280,11 +280,18 @@ public override Symbol ExpressionSymbol
 
     internal sealed partial class BoundDeconstructionAssignmentOperator : BoundExpression
     {
-        internal class AssignmentInfo
+        internal class AssignmentStep
         {
             public BoundAssignmentOperator Assignment;
-            public BoundLValuePlaceholder LValuePlaceholder;
-            public BoundRValuePlaceholder RValuePlaceholder;
+            public BoundDeconstructValuePlaceholder OutputPlaceholder;
+            public BoundDeconstructValuePlaceholder TargetPlaceholder;
+        }
+
+        internal class DeconstructStep
+        {
+            public MethodSymbol DeconstructMemberOpt; // the deconstruct member, or null if tuple deconstruction
+            public BoundDeconstructValuePlaceholder TargetPlaceholder;
```

Comment:
Never mind, since we might not create the deconstruct step in case of an error, the suggestion is not an option.

---
In reply to: [65796074](https://github.com/dotnet/roslyn/pull/11715#discussion_r65796074) [](ancestors = 65796074)